Applications/Spire/Source/Ui/DropDownList.cpp

```diff
@@ -61,18 +32,47 @@ DropDownList::DropDownList(ListView& list_view, QWidget* parent)
   m_panel->set_closed_on_blur(true);
   m_panel->layout()->itemAt(0)->widget()->setSizePolicy(QSizePolicy::Minimum,
     QSizePolicy::Preferred);
+  if(auto list_item = m_list_view->get_list_item(0)) {
+    setMaximumHeight(10 * list_item->sizeHint().height());
+    get_panel_border_size();
+    setMinimumWidth(m_panel->parentWidget()->size().width() -
+      m_panel_border_size.width());
+  }
 }
 
 bool DropDownList::event(QEvent* event) {
   if(event->type() == QEvent::ShowToParent) {
-    auto list_item = m_list_view->get_list_item(0);
-    if(list_item) {
-      setMaximumHeight(10 * list_item->sizeHint().height());
-      auto border_size = get_border_size(*m_panel);
-      setMinimumWidth(m_panel->parentWidget()->size().width() -
-        border_size.width());
+    if(m_list_view->get_list_model()->get_size() > 0) {
       m_panel->show();
     }
   }
   return QWidget::event(event);
 }
+
+void DropDownList::get_panel_border_size() {
```

Comment:
Use the style signal to get this information and keep it up-to-date. Styles can change at any moment and can also be animated.